File: app/src/main/java/com/appster/dentamatch/model/JobDetailModel.java
/*
 *
 *  * Copyright © 2018 DentaMatch. All rights reserved.
 *  * Developed by Appster.
 *  *
 *
 */

package com.appster.dentamatch.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Created by Appster on 01/02/17.
 * To inject activity reference.
 */

public class JobDetailModel {

    private int id;
    private int jobType;
    private int noOfJobs;
    private int isMonday;
    private int isTuesday;
    private int isWednesday;
    private int isThursday;
    private int isFriday;
    private int isSaturday;
    private int isSunday;
    private String templateName;
    private String templateDesc;
    private String workEverydayStart;
    private String workEverydayEnd;
    private String mondayStart;
    private String mondayEnd;
    private String tuesdayStart;
    private String tuesdayEnd;
    private String wednesdayStart;
    private String wednesdayEnd;
    private String thursdayStart;
    private String thursdayEnd;
    private String fridayStart;
    private String fridayEnd;
    private String saturdayStart;
    private String saturdayEnd;
    private String sundayStart;
    private String sundayEnd;
    @SerializedName("jobtitleName")
    private String jobTitleName;
    private String officeName;
    private String officeDesc;
    private String address;
    @SerializedName("zipcode")
    private int zipCode;
    private double latitude;
    private double longitude;
    private String createdAt;
    private int jobPostedTimeGap;
    private String officeTypeName;
    private double distance;
    //    public List<Object> jobTypeDates = null;
    private String jobTypeString;
    private int isApplied;
    private int isSaved;
    private String partTimeDays;
    private ArrayList<String> jobTypeDates;
    private String percentaSkillsMatch;
    private String matchedSkills;
    private String templateSkillsCount;
    private int payRate;

    public String getPercentaSkillsMatch() {
        return percentaSkillsMatch;
    }

    public void setPercentaSkillsMatch(String percentaSkillsMatch) {
        this.percentaSkillsMatch = percentaSkillsMatch;
    }

    public String getMatchedSkills() {
        return matchedSkills;
    }

    public void setMatchedSkills(String matchedSkills) {
        this.matchedSkills = matchedSkills;
    }

    public String getTemplateSkillsCount() {
        return templateSkillsCount;
    }

    public void setTemplateSkillsCount(String templateSkillsCount) {
        this.templateSkillsCount = templateSkillsCount;
    }


    public ArrayList<String> getJobTypeDates() {
        return jobTypeDates;
    }

    public void setIsSaved(int isSaved) {
        this.isSaved = isSaved;
    }

    public int getId() {
        return id;
    }

    public int getJobType() {
        return jobType;
    }

    public int getIsMonday() {
        return isMonday;
    }

    public int getIsTuesday() {
        return isTuesday;
    }

    public int getIsWednesday() {
        return isWednesday;
    }

    public int getIsThursday() {
        return isThursday;
    }

    public int getIsFriday() {
        return isFriday;
    }

    public int getIsSaturday() {
        return isSaturday;
    }

    public int getIsSunday() {
        return isSunday;
    }

    public String getTemplateName() {
        return templateName;
    }

    public String getTemplateDesc() {
        return templateDesc;
    }

    public String getWorkEverydayStart() {
        return workEverydayStart;
    }

    public String getWorkEverydayEnd() {
        return workEverydayEnd;
    }

    public String getMondayStart() {
        return mondayStart;
    }

    public String getMondayEnd() {
        return mondayEnd;
    }

    public String getTuesdayStart() {
        return tuesdayStart;
    }

    public String getTuesdayEnd() {
        return tuesdayEnd;
    }

    public String getWednesdayStart() {
        return wednesdayStart;
    }

    public String getWednesdayEnd() {
        return wednesdayEnd;
    }

    public String getThursdayStart() {
        return thursdayStart;
    }

    public String getThursdayEnd() {
        return thursdayEnd;
    }

    public String getFridayStart() {
        return fridayStart;
    }

    public String getSaturdayStart() {
        return saturdayStart;
    }

    public String getSaturdayEnd() {
        return saturdayEnd;
    }

    public String getSundayStart() {
        return sundayStart;
    }

    public String getSundayEnd() {
        return sundayEnd;
    }

    public String getJobTitleName() {
        return jobTitleName;
    }

    public String getOfficeName() {
        return officeName;
    }

    public String getAddress() {
        return address;
    }

    public int getZipCode() {
        return zipCode;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public int getJobPostedTimeGap() {
        return jobPostedTimeGap;
    }

    public String getOfficeTypeName() {
        return officeTypeName;
    }

    public double getDistance() {
        return distance;
    }

    public String getJobTypeString() {
        return jobTypeString;
    }

    public int getIsApplied() {
        return isApplied;
    }

    public int getIsSaved() {
        return isSaved;
    }

    public String getOfficeDesc() {
        return officeDesc;
    }

    public String getFridayEnd() {
        return fridayEnd;
    }

    public int getNoOfJobs() {
        return noOfJobs;
    }

    public int getPayRate() {
        return payRate;
    }
}

File: app/src/main/java/com/appster/dentamatch/model/JobDetailModel.kt
/*
 *
 *  * Copyright © 2018 DentaMatch. All rights reserved.
 *  * Developed by Appster.
 *  *
 *
 */

package com.appster.dentamatch.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

import java.util.ArrayList

/**
 * Created by Appster on 01/02/17.
 * To inject activity reference.
 */

@Parcelize
data class JobDetailModel(
    var id: Int = 0,
    var jobType: Int = 0,
    var noOfJobs: Int = 0,
    var isMonday: Int = 0,
    var isTuesday: Int = 0,
    var isWednesday: Int = 0,
    var isThursday: Int = 0,
    var isFriday: Int = 0,
    var isSaturday: Int = 0,
    var isSunday: Int = 0,
    var templateName: String? = null,
    var templateDesc: String? = null,
    var workEverydayStart: String? = null,
    var workEverydayEnd: String? = null,
    var mondayStart: String? = null,
    var mondayEnd: String? = null,
    var tuesdayStart: String? = null,
    var tuesdayEnd: String? = null,
    var wednesdayStart: String? = null,
    var wednesdayEnd: String? = null,
    var thursdayStart: String? = null,
    var thursdayEnd: String? = null,
    var fridayStart: String? = null,
    var fridayEnd: String? = null,
    var saturdayStart: String? = null,
    var saturdayEnd: String? = null,
    var sundayStart: String? = null,
    var sundayEnd: String? = null,
    @SerializedName("jobtitleName")
    var jobTitleName: String? = null,
    var officeName: String? = null,
    var officeDesc: String? = null,
    var address: String? = null,
    @SerializedName("zipcode")
    var zipCode: Int = 0,
    var latitude: Double = 0.toDouble(),
    var longitude: Double = 0.toDouble(),
    var createdAt: String? = null,
    var jobPostedTimeGap: Int = 0,
    var officeTypeName: String? = null,
    var distance: Double = 0.toDouble(),
    var jobTypeString: String? = null,
    var isApplied: Int = 0,
    var isSaved: Int = 0,
    var partTimeDays: String? = null,
    var jobTypeDates: ArrayList<String>? = null,
    var percentaSkillsMatch: String? = null,
    var matchedSkills: String? = null,
    var templateSkillsCount: String? = null,
    var payRate: Int = 0
) : Parcelable
